map/framework.cpp

```diff
@@ -2166,18 +2172,30 @@ std::optional<place_page::Info> Framework::BuildPlacePageInfo(
     }
   }
 
+  bool isBuildingSelected = false;
   if (isFeatureMatchingEnabled && !selectedFeature.IsValid())
+  {
     selectedFeature = FindBuildingAtPoint(buildInfo.m_mercator);
+    isBuildingSelected = selectedFeature.IsValid();
+  }
 
   bool showMapSelection = false;
   if (selectedFeature.IsValid())
   {
+    // Selection circle should match feature
     FillFeatureInfo(selectedFeature, outInfo);
-    if (buildInfo.m_isLongTap)
-      outInfo.SetMercator(buildInfo.m_mercator);
+
+    if (isBuildingSelected)
+    {
+      // Calculate distance between building centroid and tap point
+      double distancePx = (m_currentModelView.GtoP(buildInfo.m_mercator) - m_currentModelView.GtoP(outInfo.GetMercator())).Length();
+      // If user taps on a building far from it's centroid (>10px) move selection circle to tap position
+      if (distancePx > kBuildingCentroidThreshold * df::VisualParams::Instance().GetVisualScale())
```

Comment:
@vng can you please review this change?

Reply:
I think this code is not needed at all, please check https://github.com/organicmaps/organicmaps/pull/7876#discussion_r1627355744